packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -35,36 +40,32 @@ export interface SharedTreeBranchEvents<TChange> {
 export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> extends EventEmitter<
 	SharedTreeBranchEvents<TChange>
 > {
-	private head: GraphCommit<TChange>;
 	public readonly editor: TEditor;
 	private readonly transactions = new TransactionStack();
-	private readonly forks = new Set<SharedTreeBranch<TEditor, TChange>>();
 
 	/**
-	 * @param getBaseBranch - a function which retrieves the head of the base branch
-	 * @param mergeIntoBase - a function which describes how to merge this branch into the base branch which created it.
-	 * It is responsible for rebasing the changes properly across branches and updating the head of the base branch.
-	 * It returns the net change to the child branch.
-	 * @param sessionId - the session ID used to author commits made by to this branch
-	 * @param rebaser - a rebaser to rebase this branch's changes when it pulls or merges
+	 * Construct a new branch.
+	 * @param head - the head of the branch
+	 * @param sessionId - the session ID used to author commits made by this branch
+	 * @param rebaser - the rebaser used for rebasing and merging commits across branches
+	 * @param changeFamily - determines the set of changes that this branch can commit
+	 * @param anchors - an optional set of anchors that this branch will rebase whenever the branch head changes
 	 */
 	public constructor(
-		private readonly getBaseBranch: () => GraphCommit<TChange>,
-		private readonly mergeIntoBase: (forked: SharedTreeBranch<TEditor, TChange>) => TChange,
+		private head: GraphCommit<TChange>,
 		private readonly sessionId: string,
 		private readonly rebaser: Rebaser<TChange>,
-		private readonly changeFamily: ChangeFamily<TEditor, TChange>,
-		private readonly anchors: AnchorSet,
+		public readonly changeFamily: ChangeFamily<TEditor, TChange>,
+		private readonly anchors?: AnchorSet,
 	) {
 		super();
-		this.head = getBaseBranch();
 		this.editor = this.changeFamily.buildEditor(
-			(change) => this.applyChange(change, false),
-			anchors,
+			(change) => this.applyChange(change),
+			new AnchorSet(), // This branch class handles the anchor rebasing, so we don't want the editor to do any rebasing; so pass it a dummy anchor set.
```

Comment:
Shouldn't this be
```suggestion
			anchors ?? new AnchorSet(), // This branch class handles the anchor rebasing, so we don't want the editor to do any rebasing; so pass it a dummy anchor set.
```

Reply:
No; the branch does the rebasing itself in `applyChange`. Whether or not any anchors are present, we don't want the editor to do any rebasing. If the editor does the rebasing, it makes the rest of the flow in SharedTreeBranch a little grosser because the branch class still has to do the rebasing for all its other operations, just not `applyChange`.